java/core/src/java/org/apache/orc/impl/TreeReaderFactory.java

```diff
@@ -160,11 +189,152 @@ public boolean fileUsedProlepticGregorian() {
     }
   }
 
+  public static abstract class FilterContext {
```

Comment:
Unfortunately, I think this approach won't work for nested structures. I think we need to have TreeReaders only use skip and/or seek to get over the values they don't need. TreeReaders don't know about rows, only values for their column.

Reply:
As showcased in TestRowFilteringComplexTypes class only LIST and MAP types are not currently supported and will be addressed by ORC-619